packages/pluggableWidgets/line-chart-native/src/utils/SeriesLoaderHook.ts

```diff
@@ -0,0 +1,226 @@
+import { useEffect, useState } from "react";
+import { ensure } from "@mendix/pluggable-widgets-tools";
+import { Big } from "big.js";
+import { ObjectItem } from "mendix";
+
+import { SeriesType } from "../../typings/LineChartProps";
+import { LineChartDataPoints, LineChartSeries } from "../components/LineChart";
+
+interface DataSourceItemGroup {
+    groupByAttributeValue: string | boolean | Date | Big;
+    dynamicSeriesNameValue?: string;
+    dynamicStylePropertyNameValue?: string;
+    items: ObjectItem[];
+}
+
+interface DataPointsExtraction {
+    dataPoints: LineChartDataPoints;
+    xFormatter?: (xValue: number | Date) => string;
+    yFormatter?: (yValue: number | Date) => string;
+}
+
+export function useSeries(series: SeriesType[]): LineChartSeries[] | null {
+    const [chartSeries, setChartSeries] = useState<LineChartSeries[] | null>(null);
```

Comment:
Can we return empty array instead? Same as no data in this case

Reply:
There is a semantic difference between an empty array and null. If null, the loading of data failed. If empty array, no series have been configured. The latter situation is in this case not possible, because you always need to add one series.